source/dem/non_uniform_insertion.cc

```diff
@@ -122,82 +163,65 @@ NonUniformInsertion<dim>::create_random_number_container(
 }
 
 // This function assigns the insertion points of the inserted particles
-template <int dim>
-void
-NonUniformInsertion<dim>::assign_insertion_points(
-  std::vector<Point<dim>> &                    insertion_positions,
-  const Parameters::Lagrangian::InsertionInfo &insertion_information,
-  const MPI_Comm &                             communicator)
+template <>
+void NonUniformInsertion<2>::find_insertion_location_nonuniform(
+  Point<2> &                                   insertion_location,
+  const unsigned int &                         id,
+  const double &                               random_number1,
+  const double &                               random_number2,
+  const Parameters::Lagrangian::InsertionInfo &insertion_information)
+{
+  std::vector<int> insertion_index;
+  insertion_index.reserve(2);
```

Comment:
```suggestion
  insertion_index.resize(2);
```

This should be a resize here because you want to contain 3 elements and you are not pushing back. Reserve is only when you want to pre-allocate without changing size.

https://stackoverflow.com/questions/13029299/stdvectorresize-vs-stdvectorreserve

You could also use the constructor for this I guess.

Reply:
corrected.